core/src/mindustry/ui/dialogs/HostDialog.java

```diff
@@ -69,8 +69,13 @@ public void runHost(){
                 if(steam){
                     Core.app.post(() -> Core.settings.getBoolOnce("steampublic2", () -> {
                         ui.showCustomConfirm("@setting.publichost.name", "@public.confirm", "@yes", "@no", () -> {
-                            Core.settings.put("publichost", true);
-                            platform.updateLobby();
+                            ui.showCustomConfirm("@setting.publichost.name", "@public.confirm.really", "@no", "@yes", () -> {
+                                Core.settings.put("publichost", true);
+                                platform.updateLobby();
+                            }, () -> {
+                                Core.settings.put("publichost", false);
+                                platform.updateLobby();
+                            }
```

Comment:
this might fix the ci:
```suggestion
                            });
```